src/components/Button/Button.test.js

```diff
@@ -0,0 +1,43 @@
+// Button specific implementation
+// Resources https://stackoverflow.com/questions/58408178/query-a-button-with-specific-text
+
+import React from 'react';
+import { fireEvent, render, screen } from '@testing-library/react';
+import { Button } from './Button';
+
+describe('Button works as expected', () => {
+    it('Button displays the text provided', async () => {
+        render(<Button text="Simple Test" />);
+        screen.getByText(/simple test/i, { selector: 'button' });
```

Comment:
The use of RegEx is discouraged in the project you can use an string 'Simple Test' and is going to work fine